src/rules/no-named-as-default.js

```diff
@@ -15,28 +15,70 @@ module.exports = {
 
   create(context) {
     function checkDefault(nameKey, defaultSpecifier) {
+      /**
+       * For ImportDefaultSpecifier we're interested in the "local" name (`foo` for `import {bar as foo} ...`)
+       * For ExportDefaultSpecifier we're interested in the "exported" name (`foo` for `export {bar as foo} ...`)
+       */
+      const analyzedName = defaultSpecifier[nameKey].name;
+
       // #566: default is a valid specifier
-      if (defaultSpecifier[nameKey].name === 'default') { return; }
+      if (analyzedName === 'default') { return; }
 
       const declaration = importDeclaration(context, defaultSpecifier);
+      /** @type {import('../exportMap').default | null} */
+      const importedModule = ExportMapBuilder.get(declaration.source.value, context);
+      if (importedModule == null) { return; }
 
-      const imports = ExportMapBuilder.get(declaration.source.value, context);
-      if (imports == null) { return; }
+      if (importedModule.errors.length) {
```

Comment:
```suggestion
      if (importedModule.errors.length > 0) {
```

Reply:
I did not want to change the existing logic of error reporting for the part I don't fully understand. You sure this change is safe?

Reply:
Added in fd567c69499b16a4a14017a7bda2b6cb16b612e3

Reply:
oh fair, but yeah i think this would always be a safe change :-)